### PR TITLE
Issue 1073

### PR DESCRIPTION
**Pull request by andyfengHKU**
Fix issue #1073

### CODE REVIEW

File: src/processor/operator/copy_csv/copy_node_csv.cpp [hunk 1]
```diff
@@ -5,15 +5,22 @@
 namespace kuzu {
 namespace processor {
 
-string CopyNodeCSV::execute(TaskScheduler* taskScheduler, ExecutionContext* executionContext) {
-    auto nodeCSVCopier = make_unique<InMemArrowNodeCopier>(csvDescription, wal->getDirectory(),
-        *taskScheduler, *catalog, tableID, &nodesStore->getNodesStatisticsAndDeletedIDs());
-    errorIfTableIsNonEmpty(&nodesStore->getNodesStatisticsAndDeletedIDs());
-    // Note: This copy function will update the maxNodeOffset in nodesStatisticsAndDeletedIDs.
+uint64_t CopyNodeCSV::executeInternal(
+    common::TaskScheduler* taskScheduler, ExecutionContext* executionContext) {
```
Comment (acquamarin) [inline src/processor/operator/copy_csv/copy_node_csv.cpp:9]: I think you can remove the common namespace here.

File: src/processor/operator/copy_csv/copy_node_csv.cpp [hunk 2]
```diff
@@ -5,15 +5,22 @@
...
     wal->logCopyNodeCSVRecord(tableID);
-    return StringUtils::string_format("%d number of nodes has been copied to nodeTable: %s.",
-        numNodesCopied, catalog->getReadOnlyVersion()->getTableName(tableID).c_str());
+    return numNodesCopied;
+}
+
+uint64_t CopyNodeCSV::getNumTuplesInTable() {
+    // TODO(Ziyi): this chains looks weird. Fix when refactoring table statistics. Ditto in
+    // CopyRelCSV.
+    return ((TablesStatistics*)nodesStatistics)
```
Comment (acquamarin) [inline src/processor/operator/copy_csv/copy_node_csv.cpp:20]: I don't think you need to cast to `TablesStatistics*` in order to get the readOnly version.

File: src/processor/operator/copy_csv/copy_rel_csv.cpp [hunk 3]
```diff
@@ -5,17 +5,21 @@
...
-    // information in relsStatistics for this relTable.
     auto numRelsCopied = relCSVCopier->copy();
     wal->logCopyRelCSVRecord(tableID);
-    return StringUtils::string_format("%d number of rels has been copied to relTable: %s.",
-        numRelsCopied, catalog->getReadOnlyVersion()->getTableName(tableID).c_str());
+    return numRelsCopied;
+}
+
+uint64_t CopyRelCSV::getNumTuplesInTable() {
+    return ((TablesStatistics*)relsStatistics)
```
Comment (acquamarin) [inline src/processor/operator/copy_csv/copy_rel_csv.cpp:19]: Ditto, you don't need to cast this to tableStatistics.

File: src/processor/operator/copy_csv/copy_rel_csv.cpp [hunk 4]
```diff
@@ -5,17 +5,21 @@
 namespace kuzu {
 namespace processor {
 
-string CopyRelCSV::execute(TaskScheduler* taskScheduler, ExecutionContext* executionContext) {
+uint64_t CopyRelCSV::executeInternal(
+    kuzu::common::TaskScheduler* taskScheduler, ExecutionContext* executionContext) {
```
Comment (acquamarin) [inline src/processor/operator/copy_csv/copy_rel_csv.cpp:9]: Ditto, you can remove `kuzu::common`

File: src/processor/operator/copy_csv/copy_csv.cpp [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    metrics->executionTime.start();
+    errorIfTableIsNonEmpty();
+    auto numTuplesCopied = executeInternal(taskScheduler, executionContext);
+    metrics->executionTime.stop();
+    metrics->numOutputTuple.increase(numTuplesCopied);
+    return getOutputMsg(numTuplesCopied);
+}
+
+void CopyCSV::errorIfTableIsNonEmpty() {
+    auto numTuples = getNumTuplesInTable();
```
Comment (acquamarin) [inline src/processor/operator/copy_csv/copy_csv.cpp:17]: Remove `numTuples` if it is unused

**Review by acquamarin — APPROVED** [review]

File: src/processor/operator/copy_csv/copy_csv.cpp (hunk 5 above)
Comment (andyfengHKU) [reply src/processor/operator/copy_csv/copy_csv.cpp:17]: used for printing

**Pull request merged**